### PR TITLE
Merge Admin and Dashboard

### PR DESCRIPTION
**Pull request by yirano**
Merging Admin Dashboard and Headmaster Dashboard to main codebase.

### CODE REVIEW

File: src/__tests__/VillageForm.test.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+// 	render(
+// 		<Router history={history}>
+// 			<VillageForm />
+// 		</Router>
+// 	)
+
+// 	// expect(screen.getByText(/headmaster/i)).toBeInTheDocument()
+// })
+
+it('says hello', () => {
```
Comment (jidelson) [inline src/__tests__/VillageForm.test.js]: I like how you did a sanity check here.

File: src/components/App.js [hunk 2]
```diff
@@ -16,12 +18,16 @@ class App extends Component {
   render() {
     return (
       <div className="App">
-        <BrowserRouter {...this.props}>
+        <Route exact path="/login" component={Login} />
+        <Route path="/admin">
+          <AdminDashboard />
+        </Route>
+        <Route exact path="/headmaster/*">
```
Comment (Jahteo) [inline src/components/App.js:25]: love how you resolved the conflict between our branches here!

File: src/components/App.js [hunk 3]
```diff
@@ -16,12 +18,16 @@ class App extends Component {
   render() {
     return (
       <div className="App">
-        <BrowserRouter {...this.props}>
+        <Route exact path="/login" component={Login} />
+        <Route path="/admin">
+          <AdminDashboard />
+        </Route>
```
Comment (jidelson) [inline src/components/App.js:24]: Good job on refactoring this away from the landing page. (I believe that's what you did here)

File: src/components/common/Button.js [hunk 4]
```diff
@@ -1,18 +1,21 @@
...
+
+const buttonStyle = {};
+
+const Btn = styled.button`
+  border: none;
+  &:hover {
+    transition: all 200ms linear;
+  }
+  /* display: none; */
+`;
```
Comment (Jahteo) [inline src/components/common/Button.js:13]: Our reusable button! Thank you!

File: src/components/pages/Admin/AdminDashboard.js [hunk 5]
```diff
@@ -10,9 +11,17 @@ export default function AdminDashboard() {
...
       <h1>Welcome Administrator to your personal dashboard</h1>
       {/* TODO: pushing to a new route works for mobile first, but may need to be modified for better browser designs */}
-      <Button onClick={() => push('admin/libraries')}> Libraries </Button>
+      <Link to="/admin/libraries">
+        <Button>Libraries </Button>
+      </Link>
       <Button> Schools </Button>
       <Button> xxxxxx </Button>
+
+      <Switch>
```
Comment (jidelson) [inline src/components/pages/Admin/AdminDashboard.js:20]: Good job on fixing the issue that was going on here.

File: src/components/pages/School/School.component.js [hunk 6]
```diff
@@ -4,21 +4,18 @@ import { Link } from 'react-router-dom';
 import { fetchSchool } from '../../../state/actions/index';
 
 const School = props => {
-  useEffect(() => {
-    props.fetchSchool(1); // ! This doesn't work yet. we don't have a school endpoint
-  }, []);
+  console.log('School Component --> ', props);
```
Comment (Jahteo) [inline src/components/pages/School/School.component.js:7]: remove this next time around

File: src/components/pages/Admin/EditLibrary.js [hunk 7]
```diff
@@ -21,47 +21,45 @@ const initialState = {
   image: '',
 };
 
-function EditLibraryForm(props) {
+function EditLibraryForm({ editLibrary }) {
   const [formData, setFormData] = useState(initialState);
 
-  // ? Why the .libraryID?
+  const { push } = useHistory();
```
Comment (jidelson) [inline src/components/pages/Admin/EditLibrary.js:27]: Do we still need this now that you changed it below?

File: src/components/pages/Village/VillageForm.js [hunk 8]
```diff
@@ -71,7 +72,7 @@ const VillageForm = props => {
   return (
     <FormContainer>
       <Form.Item {...tailLayout}>
-        <Link to="/village">Go Back to Village Profile</Link>
+        <Link to="/school-village">Go Back to Village Profile</Link>
```
Comment (jidelson) [inline src/components/pages/Village/VillageForm.js]: Good job changing this

File: src/state/actions/index.js [hunk 9]
```diff
@@ -31,8 +38,23 @@ export const editVillage = (id, data) => () => {
...
+    .then(res => {
+      dispatch({
+        type: actionTypes.FETCH_HEADMASTER_SCHOOLS,
+        payload: res.data,
+      });
+    })
+    .catch(err => console.dir(err));
+};
+
 export const fetchSchool = id => dispatch => {
```
Comment (Jahteo) [inline src/state/actions/index.js:53]: Is this a duplicate from above?

**Review by jidelson — APPROVED** [review]

**Review by Jahteo — APPROVED** [review]
Due to time & branch conflict with feature/admin-dashboard, review was done over zoom. We checked full UI functionality of both this branches, fixed problems that had arisen and pushed. Comments on code should still be added later

File: src/state/actions/index.js (hunk 9 above)
Comment (yirano) [reply src/state/actions/index.js:53]: Nope! This one fetches the individual school and the other fetches all the school.

**Pull request merged**